### PR TITLE
Extended docker support

### PR DESCRIPTION
**Pull request by maxmeyer**
# Preface

This incorparates all changes from #27 (/cc @dexafree) rebased to current master and reduces the image size by using the multistaged build feature from docker. This also reduces the surface for attacks by reducing the amount of installed packages.

# Additions

* Multistaged docker build - minimal Alpine Linux Image for the image
* Build/Setup-scripts to make it easier for newcomers to get started
* Extend documentation about how to run the docker container

### CODE REVIEW

**Comment by dexafree** [comment]
Seems great to me.
Would you maybe consider using a `Makefile` instead of 3 different scripts for the setup?

I could help with that.

**Comment by maxmeyer** [comment]
@dexafree I thought about that too, but in the end having those scripts is more obvious than a single file - also see https://githubengineering.com/scripts-to-rule-them-all/. But I'm happy to add a simple `Makefile` which can act as another entrypoint. This is what I do in my projects normally, having both the scripts and the `Makefile`.

**Comment by maxmeyer** [comment]
@dexafree Done.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+RUN apk --no-cache add dumb-init ca-certificates
+COPY --from=builder /go/src/github.com/RadhiFadlillah/shiori/shiori /usr/local/bin/shiori
+
+WORKDIR /srv/
+RUN touch shiori.db
+
+EXPOSE 8080
+
+ENTRYPOINT ["/usr/bin/dumb-init", "--"]
```
Comment (maxmeyer) [inline Dockerfile]: @RadhiFadlillah If you wonder, what this tool is for: https://github.com/Yelp/dumb-init:

dumb-init is a simple process supervisor and init system designed to run as PID 1 inside minimal container environments (such as Docker). It is deployed as a small, statically-linked binary written in C.

**Comment by RadhiFadlillah** [comment]
Thanks @maxmeyer.

BTW, I think we don't really need to add those additional scripts, especially since `go` command is already simple enough to use. For example, `setup` is just wrapper for `go get` and `build` is just wrapper for `go build`.

File: README.md [hunk 2]
```diff
@@ -65,109 +65,231 @@ Use "shiori [command] --help" for more information about a command.
...
+# or
+make setup
+```
+
+### Build project
+
+To build the project, please run the following command.
+
+```sh
+bin/build
```
Comment (RadhiFadlillah) [inline README.md]: Here, we can just use `go build -o shiori`

File: README.md [hunk 3]
```diff
@@ -65,109 +65,231 @@ Use "shiori [command] --help" for more information about a command.
...
     ```
+## Contribute
+
+### Setup project
+
+To install dependencies required for development, please run the following
+command.
+
+```sh
+bin/setup
```
Comment (RadhiFadlillah) [inline README.md]: Here, we can just use `go get -d -v ./..`

File: README.md [hunk 4]
```diff
@@ -65,107 +65,195 @@ Use "shiori [command] --help" for more information about a command.
...
 By default, `shiori` will create database in the location where you run it. For example, if you run `shiori`. To set the database to a specific location, you can set the environment variable `ENV_SHIORI_DB` to your desired path. 
 
+## Usage with Docker
+
+There's a Dockerfile that enables you to build your own dockerized Shiori (as by now there is no official image available on Docker Hub).
+
+### Build the image
+
+```bash
+$ docker build -t shiori .
```
Comment (maxmeyer) [inline README.md]: If there's a decision for #70 this needs to be replaced by `bin/docker/build`.

**Comment by RadhiFadlillah** [comment]
Thanks a lot @maxmeyer, I will take a look at #70 later.

**Comment by maxmeyer** [comment]
Thanks @RadhiFadlillah

**Comment by maxmeyer** [comment]
@RadhiFadlillah Awesome! I just realized, that you just setup the automated docker build! :+1:

**Pull request merged**